src/SQLitePCLRaw.core/utf8z.cs

```diff
@@ -138,7 +173,19 @@ public string utf8_to_string()
                 }
             }
         }
-    }
 
+        /// <summary>
+        /// Gets the <see cref="Encoding.UTF8"/> encoded bytes for the provided <paramref name="value"/>.  The array
+        /// will include a trailing <c>\0</c> character.  The length of the array will <see cref="Encoding.UTF8"/>'s
+        /// <see cref="Encoding.GetByteCount(string)"/><c>+1</c> (for the trailing <c>\0</c> byte).  These bytes are
+        /// suitable to use with <see cref="FromSpan"/> using the extension <see
+        /// cref="MemoryExtensions.AsSpan{T}(T[])"/> or <see cref="FromPtr(byte*)"/> or <see cref="FromPtrLen(byte*,
+        /// int)"/>.  Note that for <see cref="FromPtrLen(byte*, int)"/> the length provided should not include the
+        /// trailing <c>\0</c> terminator.
+        /// </summary>
+        public static byte[] GetNullTerminatedUTF8Bytes(string value)
```

Comment:
alternate name: GetZeroTerminatedUTF8Bytes.

Reply:
I almost suggested that.  

I do somewhat prefer "zero-terminated" to "null-terminated", even though the latter is far more common usage.  I like to reserve the term "null" for pointers, references, the absence of a value in SQL, etc.  The zero-byte at the end of a C string was originally called NUL in ASCII, so at some point people spoke of a NUL-terminated string, and then NUL and null got conflated.  Usually when I go down this path I tell myself I'm being too pedantic.

Reply:
or: GetUTF8BytesWithTrailingZero

Reply:
Not bad, but it reminds me of `System.Numerics.BitOperations.TrailingZeroCount`.

I don't see a slam-dunk winner here.  Either "ZeroTerminated" or "NullTerminated" would be fine, but given my slight preference for the former, if you don't mind, perhaps I'll go ahead and change it to `GetZeroTerminatedUTF8Bytes`.

Reply:
Whatever works best for you :)